explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/counterfactual/CounterfactualExplainer.java

```diff
@@ -116,24 +137,25 @@ public static Builder builder(List<Output> goal, List<Boolean> constraints, Data
 
         final UUID problemId = UUID.randomUUID();
 
+        Function<UUID, CounterfactualSolution> initial = (UUID uuid) -> new CounterfactualSolution(entities, model, goal);
```

Comment:
It should work also without `UUID` explicit type
```suggestion
        Function<UUID, CounterfactualSolution> initial = uuid -> new CounterfactualSolution(entities, model, goal);
```